### PR TITLE
NAS-131690 / 25.04 / Allow remote audit call to return exception if unresponsive.

### PR DESCRIPTION
**Pull request by mgrimesix**
### Problem
----
An `audit.query` to the remote controller returns an empty result if the remote controller is unable to respond.  This is incorrect behavior and results in a false interpretation of the query.

### Fix
----
Raise an exception when the remote controller does not respond to an `audit.query`.

The test involves generating an audit event on the remote node.  The event is an attempt to delete a built-in user.
Updated passing test can be found [here](http://jenkins.eng.ixsystems.net:8080/job/tests/job/api_tests/1353/).

### CODE REVIEW

File: tests/api2/test_audit_basic.py [hunk 1]
```diff
@@ -115,29 +116,19 @@ def init_audit():
...
-            }],
-            {'raise_connect_error': False, 'timeout': 2, 'connect_timeout': 2}
-        )
-        yield name
-    finally:
-        if user_id is not None:
-            call('failover.call_remote', 'user.delete', [user_id])
+        # This _should_ fail.  Assert if not!
+        call('failover.call_remote', event, [user['id']])
+        assert True, "!!! Should not be allowed to delete a built-in user !!!"
```
Comment (themylogin) [inline tests/api2/test_audit_basic.py:129]: `assert True` will do nothing, what is this supposed to do?

File: tests/api2/test_audit_basic.py [hunk 2]
```diff
@@ -324,47 +315,58 @@ def test_audit_timestamps(self, svc):
...
-
-        assert tries > 0, "Failed to get expected audit entry"
-        assert remote_audit_entry != []
-        params = remote_audit_entry[0]['event_data']['params'][0]
-        assert params['username'] == name
-
-    def test_audit_ha_export(self, standby_user):
+    @pytest.mark.parametrize('remote_available', [True, False])
+    def test_audit_ha_query(self, standby_audit_event, remote_available):
+        assert standby_audit_event, "Missing event data"
```
Comment (themylogin) [inline tests/api2/test_audit_basic.py:320]: A fixture that fails to set up will fail all tests that depend on it

File: tests/api2/test_audit_basic.py (hunk 1 above)
Comment (mgrimesix) [reply tests/api2/test_audit_basic.py:129]: The goal is to generate an audit entry on the remote node.  To generate the entry I'm attempting to delete a built-in user. 
OK, let me refactor this.

File: tests/api2/test_audit_basic.py (hunk 2 above)
Comment (mgrimesix) [reply tests/api2/test_audit_basic.py:320]: Refactoring.

**Review by themylogin — APPROVED** [review]
Excellent, thank you!

**Pull request merged**